src/Iact1dUnbinnedLkl.cc

```diff
@@ -889,6 +889,78 @@ Int_t Iact1dUnbinnedLkl::ReaddNdESignal(TString filename)
   return status;
 }
 
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+// 
+// Read multiple dN_i/dE for signal from file in
+// the Segue Stereo input format produced by Jelena
+// and their corresponding branching ratios BR_i.
+// Then perform a linear combination
+// dN/dE = sum(BR_i * dN_i/dE).
+// Replacement of existing file is allowed
+// Return 0 in case of success
+//        1 if file is not found
+//
+Int_t Iact1dUnbinnedLkl::ReaddNdESignal(Int_t nFiles, TString* filenames, Float_t* branchingRatios)
+{
+  // Initialize the array of pointers and other variables
+  TFile* dNdESignalFile[nFiles];
+  TH1F*  hProvdNdESignal[nFiles];
+  TH1F*  hProvdNdESignalTemp;
+  TH1F*  linCHdNdESignal;
+
+  Int_t status      = 0;
+  Int_t nBins       = 0;
+  Double_t lowEdge  = 0.0;
+  Double_t highEdge = 0.0;
+  // open file and look for histograms
+  for(Int_t iFile=0;iFile<nFiles;iFile++)
+    {
+      dNdESignalFile[iFile]  = new TFile(filenames[iFile]);
+      hProvdNdESignalTemp = (TH1F*) dNdESignalFile[iFile]->Get("hdNdE");
+      // check that dN/dE histograms are found
+      if(!hProvdNdESignalTemp) status=1;
+      // store the histogram binning from the first historgram
+      if(nBins==0)
```

Comment:
You don't need this, the output histos from readAndInterpolate should all have fNFineBins bins, from fFineLEMin to fFineLEMax (see other calls to the function). These values can be define by the user, but have sensible default values too, so you don't need to worry in your function to recompute them

Reply:
Ok!